File: problem-solving/baekjoon/9625_BABBA.py
# https://www.acmicpc.net/problem/9625
import sys

def count_ab(k):
    a, b = 1, 0
    for i in range(k):
        prev_a, prev_b = a, b
        a += (prev_b - prev_a)
        b += prev_a
    return a, b

if __name__ == "__main__":
    k = int(sys.stdin.readline())
    a, b = count_ab(k)
    print("{} {}".format(a, b))

File: problem-solving/baekjoon/9625_BABBA.go
// https://www.acmicpc.net/problem/9625
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var k int
	fmt.Fscanln(reader, &k)
	a, b := countAB(k)
	fmt.Fprintf(writer, "%d %d\n", a, b)
}

func countAB(k int) (a, b int) {
	a = 1
	for i := 0; i < k; i++ {
		prevA, prevB := a, b
		a += prevB - prevA
		b += prevA
	}
	return a, b
}
